# Linear models

## Naoya's book

Naoya read a book cover to cover in a single session, at a rate of $55$ pages per hour. After reading for $4$ hours, he had $330$ pages left to read.

**How many pages is the book?**

**How many hours did it take Naoya to read the entire book?**

Naoya read $55$ pages per hour, so he read $55T$ pages in $T$ hours.

The total number of pages is comprised of the number of pages Naoya had already read and the number of pages that remained. We can express this with the equation $B=55T+R$, where:

* $B$ represents the book's length
* $T$ represents the time (in hours)
* $R$ represents the number of remaining pages to read at a given time

In [2]:
from sympy import *
B, T, R = symbols('B T R')
eq = Eq(B, 55 * T + R)

eq.subs([(T, 4), (R, 330)])

Eq(B, 550)

We know that after reading for $4$ hours $(T={4})$, Naoya had $330$ remaining pages to read $(R={330})$. Let's plug these values into the equation to find the value of $B$.

$\qquad B=55\cdot{4}+{330}=550$

Therefore, the book is $550$ pages long.

In [15]:
solveset(eq.subs([(B, 550), (R, 0)]))

{10}

To find how long it took Naoya to finish the entire book, we can plug $R=0$ into the equation and solve for $T$.

$\qquad\begin{aligned}550&=55T+0\\ 55T&=550\\ T&=10\end{aligned}$

The book is $550$ pages long.
It took Naoya $10$ hours to read the entire book.

## Charging the battery

A battery was charged. When the charging began, it was $23$ percent full. After $30$ minutes of charging, the battery was $89$ percent full.

**What percent was the battery charged per minute?**

**How many minutes did it take the battery to be fully charged?**

Let's say that the percentage of the battery's capacity that is charged increases by $V$ percent each minute. Then, it increases by $V\cdot T$ percent in $T$ minutes. In addition, we know that when the charging began, the battery was $23$ percent full.

The percentage of the battery's capacity that was charged at any given time is found by taking the percentage the battery was already charged when the charging began and adding to it the percentage that was additionally charged since then. We can express this with the equation $C=23+V\cdot T$, where:

* $C$ represents the percentage of the battery's capacity that is charged at a given time
* $V$ represents the charging rate (in percent per minute)
* $T$ represents the time (in minutes)

In [18]:
C, V, T = symbols('C V T')
eq = Eq(C, 23 + V * T)

We know that after $30$ minutes $(T={30})$, the battery was $89$ percent full $(C={89})$. Let's plug these values into the equation to find the value of $V$.

In [21]:
solveset(eq.subs([(C, 89), (T, 30.0)]))

{2.2}

$\qquad \begin{aligned}{89}&=23+V\cdot{30}\\ 30V&=66\\ V&=2.2\end{aligned}$

Therefore, the battery was charged at a rate of $2.2$ percent per minute.

To find how long it took the battery to be fully charged, we can plug $C=100$ into the equation and solve for $T$.

In [25]:
solveset(eq.subs([(C, 100), (V, 2.2)]))

{35.0}

$\qquad\begin{aligned}100&=23+2.2T\\ 2.2T&=77\\ T&=35\end{aligned}$

* **The battery was charged at a rate of $2.2$ percent per minute.**

* **It took the battery $35$ minutes to be fully charged.**

## Vlad and his mighty oak

When Vlad moved to his new home a few years ago, there was a young oak tree in his backyard. He measured it once a year and found that it grew by $26$ centimeters each year. $4.5$ years after he moved into the house, the tree was $292$ centimeters tall.

**How many centimeters tall was the tree when Vlad moved into the house?**
 
**How many years passed from the time Vlad moved in until the tree was $357$ centimeters tall?**

The tree grew by $26$ centimeters each year, so it grew by $26T$ centimeters in $T$ years.

The tree's height at a given time is found by taking its height when Vlad moved to his new home and adding to it the length it grew since then. We can express this with the equation $H=A+26T$, where:

* $H$ represents the tree's height at a given time (in centimeters)
* $A$ represents the tree's original height (in centimeters)
* $T$ represents the time (in years)

In [27]:
H, A, T = symbols('H A T')
eq = Eq(H, A + 26 * T)

We want to find $A$, so let's first solve the equation for $A$:

In [28]:
solveset(eq, A)

{H - 26*T}

$\qquad\begin{aligned}H&=A+26T\\ A&=H-26T\end{aligned}$

Now, we know that after $4.5$ years $(T={4.5})$, the tree was $292$ centimeters tall $(H={292})$. Let's plug these values into the equation to find the value of $A$.

In [29]:
solveset(eq, A).subs([(H, 292), (T, 4.5)])

{175.0}

$\qquad A={292}-26\cdot{4.5}=175$

Therefore, when Vlad moved into the house, the tree was $175$ centimeters tall.

To find how long it took the tree to reach a height of $357$ centimeters, we can plug $H=357$ into the equation and solve for $T$.

In [32]:
solveset(eq.subs([(H, 357), (A, 175)]), T)

{7}

$\qquad\begin{aligned}175&=357-26T\\ 26T&=182\\ T&=7\end{aligned}$

* **When Vlad moved into the house, the tree was $175$ centimeters tall**

* **The tree reached a height of $357$ centimeters $7$ years after Vlad moved in.**

## Sean's slushie

Sean tried to drink a slushy as fast as he could. He drank the slushy at a constant rate. There were originally $275$ milliliters of slushy in the cup. After $13$ seconds, $210$ milliliters of slushy remained.

**How many milliliters per second did Sean drink?**

**How many seconds did it take Sean to drink all the slushy?**

In [4]:
y2, y1, x2, x1 = symbols('y2 y1 x2 x1')
x1 = 0
y1 = 275
x2 = 13
y2 = 210
m = (y2 - y1)/(x2 - x1)
m

-5.0

In [5]:
275/5

55.0

## Agent Hunt's files

Agent Hunt transferred classified files from the CIA mainframe onto his flash drive. The drive had some files on it before the transfer, and the transfer happened at a rate of $4.4$ megabytes per second. After $32$ seconds, there were $384$ megabytes on the drive. The drive had a maximum capacity of $1000$ megabytes.

**How many megabytes full was the drive when the transfer began?**

The files were transferred at a rate of $4.4$ megabytes per second, so $4.4T$ megabytes were transferred in $T$ seconds.

The size of the files on the drive is comprised of the files that were on the drive before the transfer began and the files that were transferred. We can express this with the equation $S=A+4.4T$, where:

* $S$ represents the size of the files on the drive at a given time (in megabytes)

* $A$ represents the size of the files that were on the drive before the transfer began (in megabytes)

* $T$ represents the time (in seconds)

In [7]:
S, A, T = symbols('S A T')
eq = Eq(S, A + 4.4 * T)

We want to find $A$, so let's first solve the equation for $A$:

In [8]:
solveset(eq, A)

{S - 4.4*T}

$\qquad\begin{aligned}S&=A+4.4T\\ A&=S-4.4T\end{aligned}$

Now, we know that after $32$ seconds $(T={32})$, there were $384$ megabytes on the drive $(S={384})$. Let's plug these values into the equation to find the value of $A$.

In [10]:
eq = Eq(A, S - 4.4 * T)
eq.subs([(T, 32), (S, 384)])

Eq(A, 243.2)

$\qquad A={384}-4.4\cdot{32}=243.2$

Therefore, when the transfer began, the drive had $243.2$ megabytes on it.

**How many seconds from the time that Agent Hunt started the transfer did it take the drive to be completely full?**

Plug $S=1000$ into the equation and solve for $T$:

In [14]:
solveset(eq.subs([(A, 243.2), (S, 1000)]))

{172.0}

$\qquad\begin{aligned}243.2&=1000-4.4T\\ 4.4T&=756.8\\ T&=172\end{aligned}$

**It took the drive $172$ seconds to be completely full.**

## Jada's cumin

Jada has a stand in the marketplace where she sells ground cumin. Her weekly expenses are $\$300$, and she sells her cumin at a fixed price per kilogram. If she sells $120$ kilograms of cumin, her weekly net profit is $\$600$.

**How many dollars per kilogram is Jada's cumin?**

Let's say that each kilogram of cumin costs $C$ dollars. Then, $W$ kilograms of cumin cost $W\cdot C$ dollars. In addition, we know that Jada's weekly expenses are $\$300$.

Jada's weekly net profit is found by taking her income from selling cumin and subtracting her weekly expenses. We can express this with the equation $P=W\cdot C-300$, where:

* $P$ represents Jada's weekly net profit (in dollars)
* $W$ represents the amount of cumin she sold (in kilograms)
* $C$ represents the price of $1$ kilogram of cumin (in dollars)

In [1]:
from sympy import *
P, W, C = symbols('P W C')
eq = Eq(P, W * C - 300)

We know that if Jada sells $120$ kilograms of cumin $(W={120})$ her weekly net profit is $\$600$ $(P=600)$. Let's plug these values into the equation to find the value of $C$.

In [2]:
solveset(eq.subs([(P, 600), (W, 120)]))

{15/2}

$\qquad \begin{aligned}{600}&={120}\cdot C-300\\ 120C&=900\\ C&=7.5\end{aligned}$

Therefore, each kilogram of Jada's cumin costs $\$7.50$.

**How many kilograms of cumin does Jada have to sell to cover her weekly expenses?**

Plug $P=0$ into the equation and solve for $W$:

In [9]:
solveset(eq.subs([(P, 0), (C, 7.5)]), W)

{40.0}

$\qquad\begin{aligned}0&=W\cdot7.5-300\\ 7.5W&=300\\ W&=40\end{aligned}$

* **Each kilogram of Jada's cumin costs $\$7.50$.**

* **Jada has to sell $40$ kilograms of cumin to cover her weekly expenses.**

## Alexandra's cumin

Alexandra has a stand in the marketplace where she sells ground cumin. She has a fixed amount of weekly expenses, and she sells her cumin for $\$7.20$ per kilogram. If she sells $90$ kilograms of cumin, her weekly net profit is $\$414$.

**What are Alexandra's weekly expenses?**

Each kilogram of Alexandra's cumin costs $\$7.20$ so Alexandra makes a profit of $7.2W$ dollars for selling $W$ kilograms of cumin.

Alexandra's weekly net profit is found by taking her income from selling cumin and subtracting from it her weekly expenses. We can express this with the equation $P=7.2W-E$, where:

* $P$ represents Alexandra's weekly net profit (in dollars)
* $W$ represents the amount of cumin she sold (in kilograms)
* $E$ represents Alexandra's expenses (in dollars)

In [14]:
from sympy import *
P, W, E = symbols('P W E')
eq = Eq(P, 7.2 * W - E)

We want to find $E$, so let's first solve the equation for $E$:

In [15]:
solveset(eq, E)

{-P + 7.2*W}

$\qquad\begin{aligned}P&=7.2W-E\\ E&=7.2W-P\end{aligned}$

Now, we know that if Alexandra sells $90$ kilograms of cumin $(W={90})$, her weekly net profit is $\$414$ $(P={414})$. Let's plug these values into the equation to find the value of $E$.

In [16]:
solveset(eq, E).subs([(W, 90), (P, 414)])

{234.0}

$\qquad E=7.2\cdot{90}-{414}=234$

Therefore, Alexandra's weekly expenses are $\$234$.

**How many kilograms of cumin does Alexandra have to sell to cover her weekly expenses?**

To find how much cumin Alexandra has to sell to cover her weekly expenses, we can plug $P=0$, into the equation and solve for $W$.

In [18]:
solveset(eq, W).subs([(E, 234), (P, 0)])

{32.5}

$\qquad\begin{aligned}234&=7.2W-0\\ 7.2W&=234\\ W&=32.5\end{aligned}$

* **Alexandra's weekly expenses are $\$234$.**

* **Alexandra has to sell $32.5$ kilograms of cumin to cover her weekly expenses.**

## William's marbles

William has a $26$-liter glass tank. First, he wants to put some marbles in it, all of the same volume. Then, he wants to fill the tank with water until it's completely full. If he uses $85$ marbles, he will have to add $20.9$ liters of water.

**How many liters is each marble?**

Let's say that the volume of a single marble is $V$ liters. Then, the volume of $N$ marbles is $N\cdot V$ liters. In addition, we know that the volume of the tank is $26$ liters.

The sum of the water's volume and the marbles' volume should be equal to the tank's volume. We can express this with the equation $W+N\cdot V=26$, where:
* $W$ represents the volume of water used (in liters)
* $N$ represents the number of marbles used
* $V$ represents the volume of a single marble (in liters)

In [19]:
from sympy import *
W, N, V = symbols('W N V')
eq = Eq(W + N * V, 26)

We know that if William uses $85$ marbles $(N={85})$, he will have to add $20.9$ liters of water $(W={20.9})$. Let's plug these values into the equation to find the value of $V$:

In [21]:
solveset(eq.subs([(N, 85), (W, 20.9)]))

{0.06}

$\qquad \begin{aligned}{20.9}+{85}V&=26\\ 85V&=5.1\\ V&=0.06\end{aligned}$

Therefore, the volume of each marble is $0.06$ liters.

**How many liters of water is necessary if William uses $200$ marbles?**

Plug $N=200$ into the equation and solve for $W$:

In [25]:
solveset(eq.subs(N, 200), W)

{-200*V + 26}

$\qquad\begin{aligned}W+200\cdot 0.06&=26\\ W&=26-12\\ W&=14\end{aligned}$

* **The volume of each marble is $0.060$ liters.**

* **William needs $14$ liters of water if he uses $200$ marbles.**

## Zane's volcano climbing

Zane is a dangerous fellow who likes to go rock climbing in active volcanoes. One time while inside a volcano, he heard some rumbling, so he decided to climb up out of there as quickly as he could. He climbed up at a rate of $4$ meters per second. After $3$ seconds, he was $13$ meters below the edge of the volcano.

**How many meters was Zane below the edge of the volcano when he started climbing?**

Zane climbed up at a rate of $4$ meters per second, so he climbed up $4T$ meters in $T$ seconds.

The remaining distance Zane had to climb is found by taking the original distance and subtracting from it the distance Zane had already climbed. We can express this with the equation $R=D-4T$, where:
* $R$ represents Zane's remaining distance to climb (in meters)
* $D$ represents Zane's original distance from the edge of the volcano (in meters)
* $T$ represents the time (in seconds)

In [26]:
from sympy import *
R, D, T = symbols('R D T')
eq = Eq(R, D - 4 * T)

We want to find $D$, so let's first solve the equation for $D$:

In [27]:
solveset(eq, D)

{R + 4*T}

$\qquad\begin{aligned}R&=D-4T\\ D&=R+4T\end{aligned}$

Now, we know that after $3$ seconds $(T={3})$, Zane was $13$ meters below the edge of the volcano $(R={13})$. Let's plug these values into the equation to find the value of $D$.

In [28]:
solveset(eq.subs([(T, 3), (R, 13)]), D)

{25}

$\qquad D={13}+4\cdot{3}=25$

Therefore, Zane was $25$ meters below the edge of the volcano when he started climbing.

**In total, how many seconds did it take Zane to reach the edge of the volcano?**

Plug $R=0$ into the equation and solve for $T$.

In [31]:
solveset(eq.subs([(R, 0), (D, 25.0)]), T)

{6.25}

$\qquad\begin{aligned}25&=0+4T\\ 4T&=25\\ T&=6.25\end{aligned}$

* **Zane was $25$ meters below the edge of the volcano when he started climbing.**

* **It took Zane $6.25$ seconds to reach the edge of the volcano.**

## Shryia's book reading

Shryia read a $481$-page-long book cover to cover in a single session, at a constant rate. After reading for $1.5$ hours, she had $403$ pages left to read.

**How many pages per hour was Shryia reading?**

Let's say that Shryia was reading at a rate of $V$ pages per hour. Then, she read $V\cdot T$ pages in $T$ hours. In addition, we know that she had $481$ pages to read in total.

The total number of pages is comprised of the number of pages Shryia had already read and the number of pages that remained. We can express this with the equation $481=V\cdot T+R$, where:
* $V$ represents Shryia's reading speed (in pages per hour)
* $T$ represents the time (in hours)
* $R$ represents the number of remaining pages to read at a given time

In [32]:
from sympy import *
V, T, R = symbols('V T R')
eq = Eq(481, V * T + R)

We know that after $1.5$ hours $(T={1.5})$, Shryia had $403$ pages left to read $(R={403})$. Let's plug these values into the equation to find the value of $V$.

In [34]:
solveset(eq.subs([(T, 1.5), (R, 403)]))

{52.0}

$\qquad \begin{aligned}481&=V\cdot{1.5}+{403}\\ 1.5V&=78\\ V&=52\end{aligned}$

Therefore, Shryia was reading at a rate of $52$ pages per hour.

**How many hours did it take her to read the entire book?**

To find how long it took Shryia to finish the entire book, we can plug $R=0$ into the equation and solve for $T$.

In [36]:
solveset(eq.subs([(R, 0), (V, 52.0)]))

{9.25}

$\qquad\begin{aligned}481&=52T+0\\ 52T&=481\\ T&=9.25\end{aligned}$

* **Shryia was reading at a rate of $52$ pages per hour.**

* **It took Shryia $9.25$ hours to finish the entire book.**

## Andrei's tank of marbles

Andrei has a glass tank. First, he wants to put in some marbles, each of which has a volume of $0.04$ liters. Then, he wants to fill the tank with water until it's completely full. If he uses $50$ marbles, he will have to add $33$ liters of water.

**How many liters is the volume of the tank?**

The volume of each marble is $0.04$ liters, so the volume of $N$ marbles is $0.04N$ liters.

The sum of the water's volume and the marbles' volume should be equal to the tank's volume. We can express this with the equation $W+0.04N=T$, where:
* $W$ represents the volume of water used (in liters)
* $N$ represents the number of marbles used
* $T$ represents the tank's volume (in liters)

In [37]:
from sympy import *
W, N, T = symbols('W N T')
eq = Eq(W + 0.04 * N, T)

We know that if Andrei uses $50$ marbles $(N={50})$, he will have to add $33$ liters of water $(W={33})$. Let's plug these values into the equation to find the value of $T$.

In [39]:
solveset(eq.subs([(N, 50), (W, 33)]))

{35.0}

$\qquad T={33}+0.04\cdot{50}=35$

Therefore, the volume of the tank is $35$ liters.

**Andrei has exactly $20$ liters of water. How many marbles does he need so the tank is full?**

To find the number of marbles that corresponds to $20$ liters of water, we can plug $W=20$ into the equation and solve for $N$.

In [40]:
solveset(eq.subs([(T, 35), (W, 20)]))

{375.0}

$\qquad\begin{aligned}35&=20+0.04N\\ 0.04N&=15\\ N&=375\end{aligned}$

* **The volume of the tank is $35$ liters.**

* **Andrei needs $375$ marbles to fill up the tank with $20$ liters of water.**

## Wang Yong's debt payment

Wang Yong owes the bank $\$8500$. To repay the debt, he paid a fixed amount back to the bank each month. After $12$ months, his remaining debt was $\$6460$.

**How much did Wang Yong pay each month?**

**How many months did it take Wang Yong to pay back his entire debt?**

Let's say that Wang Yong pays $V$ dollars back to the bank each month. Then, he pays $V\cdot T$ dollars in $T$ months. In addition, we know that his original debt was $\$8500$.

Wang Yong's remaining debt is found by taking the original debt and subtracting from it the money he already paid back. We can express this with the equation $R=8500-V\cdot T$, where:
* $R$ represents Wang Yong's remaining debt at a given time (in dollars)
* $V$ represents Wang Yong's monthly payment to the bank(in dollars)
* $T$ represents the time (in months)

In [41]:
from sympy import *
R, V, T = symbols('R V T')
eq = Eq(R, 8500 - V * T)

We know that after $12$ months $(T={12})$, Wang Yong's remaining debt was $\$6460$ $(R={6460})$. Let's plug these values into the equation to find the value of $V$.

In [42]:
solveset(eq.subs([(T, 12), (R, 6460)]))

{170}

$\qquad \begin{aligned}{6460}&=8500-V\cdot{12}\\ 12V&=2040\\ V&=170\end{aligned}$

Therefore, Wang Yong paid the bank $\$170$ each month.

To find how long it took Wang Yong to pay back his entire debt, we can plug $R=0$ into the equation and solve for $T$.

In [43]:
solveset(eq.subs([(R, 0), (V, 170)]))

{50}

$\qquad\begin{aligned}0&=8500-170T\\ 170T&=8500\\ T&=50\end{aligned}$

* **Wang Yong paid the bank $\$170$ each month.**

* **It took Wang Yong $50$ months to pay back his entire debt.**